File: bindings/python/examples/client/09_transaction.py
from iota_sdk import Client, MnemonicSecretManager, AddressAndAmount
from dotenv import load_dotenv
import os

load_dotenv()

node_url = os.environ.get('NODE_URL', 'https://api.testnet.shimmer.network')

# Create a Client instance
client = Client(nodes=[node_url])

if 'MNEMONIC' not in os.environ:
    raise Exception(".env MNEMONIC is undefined, see .env.example")

secret_manager = MnemonicSecretManager(
    os.environ['MNEMONIC'])

address_and_amount = AddressAndAmount(
    'rms1qzpf0tzpf8yqej5zyhjl9k3km7y6j0xjnxxh7m2g3jtj2z5grej67sl6l46',
    1000000,
)

# Create and post a block with a transaction
block = client.build_and_post_block(secret_manager, output=address_and_amount)
print(f'Block sent: {os.environ["EXPLORER_URL"]}/block/{block[0]}')

File: bindings/nodejs/examples/client/09-transaction.ts
// Copyright 2021-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, SecretManager, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./client/09-transaction.ts

// In this example we will send a transaction.
async function run() {
    initLogger();
    for (const envVar of ['NODE_URL', 'MNEMONIC', 'EXPLORER_URL']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL as string],
        localPow: true,
    });

    try {
        // Configure your own mnemonic in ".env". Since the output amount cannot be zero, the mnemonic must contain non-zero
        // balance
        const secretManager = {
            mnemonic: process.env.MNEMONIC as string,
        };

        // We generate an address from our own mnemonic so that we send the funds to ourselves
        const addresses = await new SecretManager(
            secretManager,
        ).generateEd25519Addresses({
            range: {
                start: 1,
                end: 2,
            },
        });

        // We prepare the transaction
        // Insert the output address and amount to spend. The amount cannot be zero.
        const blockIdAndBlock = await client.buildAndPostBlock(secretManager, {
            output: {
                address: addresses[0],
                amount: BigInt(1000000),
            },
        });

        console.log(
            `Block sent: ${process.env.EXPLORER_URL}/block/${blockIdAndBlock[0]}`,
        );
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
